Mathlib/RingTheory/Localization/BaseChange.lean

```diff
@@ -49,6 +49,39 @@ theorem isLocalizedModule_iff_isBaseChange : IsLocalizedModule S f ↔ IsBaseCha
     LinearEquiv.restrictScalars_apply, LinearEquiv.trans_apply, IsBaseChange.equiv_symm_apply,
     IsBaseChange.equiv_tmul, one_smul]
 
+namespace IsLocalization
+
+include S
+open TensorProduct Algebra.TensorProduct
+
+variable (M₁ M₂) [AddCommMonoid M₁] [AddCommMonoid M₂] [Module R M₁] [Module R M₂]
+  [Module A M₁] [Module A M₂] [IsScalarTower R A M₁] [IsScalarTower R A M₂]
+
+theorem tensorProduct_compatibleSMul : CompatibleSMul R A M₁ M₂ where
+  smul_tmul a _ _ := by
+    obtain ⟨r, s, rfl⟩ := IsLocalization.mk'_surjective S a
+    rw [← (map_units A s).smul_left_cancel]
+    simp_rw [algebraMap_smul, smul_tmul', ← smul_assoc, smul_tmul, ← smul_assoc, smul_mk'_self,
+      algebraMap_smul, smul_tmul]
+
+noncomputable example : M₁ ⊗[A] M₂ ≃ₗ[A] M₁ ⊗[R] M₂ :=
+  have := tensorProduct_compatibleSMul S A M₁ M₂
+  equivOfCompatibleSMul R M₁ M₂ A
+
+/-- If A is a localization of a commutative ring R, the tensor product of A with A over R is
+canonically isomorphic as A-algebras to A itself. -/
+noncomputable def tensorSelfAlgEquiv : A ⊗[R] A ≃ₐ[A] A :=
```

Comment:
Do we also have a version of this for modules? E.g. take a Q-vector space, and tensor it over Z with Q, then you get the same vector space.

Reply:
I added this as [an example](https://github.com/leanprover-community/mathlib4/pull/18744/commits/122e2c73e3564888406329e68a5daab6ca22e793#diff-9e7f4ae32a2e11bf6b9c3cd40f4af9f660c7adf15dd698438cc2b54796410b6aR71-R73) and slightly refactored `tensorSelfAlgEquiv`.